### PR TITLE
docs(running renovate): rewrite

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Rewrite:
  - Use simple words and phrases
  - Create sections for Docker Hub full and slim images
  - Re-order some text to put the important/most used stuff up front

## Context

Drive by PR, not fixing any issue with this.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [inline docs/usage/getting-started/running.md:48]: Docker Hub has a `-slim` tag for the "slim" images, but I can't find a `-full` tag for the "full" images. Is that intended?

I think we mean that the full/default image has all the stuff included. But it's a bit confusing to call it "full" when there's no `-full` tag on Docker Hub.

What do you think?

Comment (rarkins) [reply docs/usage/getting-started/running.md:48]: Yes, if there's no suffix then it means it's full. Maybe we can say the default image

File: docs/usage/getting-started/running.md
Comment (rarkins) [inline docs/usage/getting-started/running.md:53]:
```suggestion
- Comes pre-installed _one_ version of each language/manager - usually the latest
```

File: docs/usage/getting-started/running.md
Comment (rarkins) [inline docs/usage/getting-started/running.md:195]:
```suggestion
If you are running on any platform except github.com, you should also set the environment variable `GITHUB_COM_TOKEN` and put the Personal Access Token for github.com in it.
```

No, it's common - and in some cases best - practice to put secrets in environment variables

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/getting-started/running.md
Comment (viceice) [inline docs/usage/getting-started/running.md:49]: i think this is now wrong. as both images are using install mode by default, so they always install each manager on the fly. they are only skipped when the same version is already installed.

so I would prefer to suggest the slim image.

**Review by viceice — DISMISSED** [review]

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [inline docs/usage/getting-started/running.md:53]:
```suggestion
- Comes pre-installed with _one_ version of each language/manager - usually the latest
```

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [reply docs/usage/getting-started/running.md:49]: This means the pros/cons of the `slim` and default images change. It sounds like both images install the needed managers on the fly.

What are the tradeoffs that you make when you choose `slim` vs default image? I imagine getting the `slim` image, and then downloading 20 package managers is worse than getting the full fat default image, for example?

Can you comment what kind of benefits/drawbacks each choice has? I'll then edit the docs to put those comparisons in the docs. 😄

Comment (rarkins) [reply docs/usage/getting-started/running.md:49]: The only significant reason for the full image is for people who don't want to download/install things at runtime?

Comment (viceice) [reply docs/usage/getting-started/running.md:49]: sure, bit they also need to change to global mode. otherwise renovate will still install latest version of the tools or the requested version from repo. and the preinstalled tool are sometimes outdated if the user isn't updating the docker image regularly.


we also support a persistent cache for downloaded tools, so that we only need to unpack on second+ run.

File: docs/usage/getting-started/running.md
Comment (viceice) [inline docs/usage/getting-started/running.md:39]: maybe we should add a note that we only support `linux/amd64` (and `linux/arm64` soon). so no windows container or any other platform.

in general renovate supports also running on windows with global mode

File: docs/usage/getting-started/running.md
Comment (rarkins) [reply docs/usage/getting-started/running.md:49]: Maybe the full image should default to binarySource=global as a breaking change in a next major release?

Comment (viceice) [reply docs/usage/getting-started/running.md:49]: 🤷‍♂️

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [reply docs/usage/getting-started/running.md:39]:
```suggestion
These Docker images work on all the hosting platforms that Renovate supports.

You must run the Docker images in a `linux/amd64` container.
We're working on `linux/arm64` container support.
You can't run the Docker images on a Windows or macOS container.

In general, you can run Renovate on Windows with global mode.
```

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [inline docs/usage/getting-started/running.md:45]: First we say:

> You can't run the Docker images on a Windows or macOS container.

And then:

> In general, you can run Renovate on Windows with global mode.

I think we need more explanation for the "run Renovate on Windows with global mode" part. For example:

- Explain what global mode does, or link to the relevant section of the docs
- Be clear what version of Renovate you can run on Windows
- Any other thing that I'm forgetting, or don't know about 😄

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [reply docs/usage/getting-started/running.md:55]: > so I would prefer to suggest the slim image.

> The only significant reason for the full image is for people who don't want to download/install things at runtime?

Should we make `-slim` the default version you get when you use the `latest` tag then?

Comment (rarkins) [reply docs/usage/getting-started/running.md:55]: I don't think we should be holding up this PR while discussing such a big change like that

File: docs/usage/getting-started/running.md
Comment (viceice) [reply docs/usage/getting-started/running.md:45]: you can run renovate native on windows and macos via npm package. then you need to install required tool manually.

i don't think that there are macos containers. but windows container which run windows inside docker container.

you can run renovate docker images in docker desktop. on windows you need to use wsl. macos will use the `linux/amd64` even on modern arm mac's (changes soon with v36).

File: docs/usage/getting-started/running.md
Comment (viceice) [inline docs/usage/getting-started/running.md:67]: it installs required tools on the fly by default.
it optionally can use docker to install tools inside sidecar image instead of the current image.

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [reply docs/usage/getting-started/running.md:55]: Here's the issue to discuss my idea: https://github.com/renovatebot/docker-renovate/issues/1237

Comment (HonkingGoose) [reply docs/usage/getting-started/running.md:62]: > Maybe the full image should default to binarySource=global as a breaking change in a next major release?

Here's the issue for that idea: 😄 

- https://github.com/renovatebot/docker-renovate-full/issues/1599

File: docs/usage/getting-started/running.md
Comment (viceice) [inline docs/usage/getting-started/running.md:57]: link to binary source? because that needs to be set to docker too

File: docs/usage/getting-started/running.md
Comment (viceice) [inline docs/usage/getting-started/running.md:61]:
```suggestion
The default image comes with most package manager that Renovate supports.
```
some of the latest are missing

File: docs/usage/getting-started/running.md
Comment (viceice) [inline docs/usage/getting-started/running.md:67]: should we link to some extended description on containerbase?

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [inline docs/usage/getting-started/running.md:61]:
```suggestion
The default image comes with most package managers that Renovate supports, but not _all_ package managers.
```

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [reply docs/usage/getting-started/running.md:57]:
```suggestion
If you want, you can map the Docker socket into the container so that Renovate can dynamically invoke "sidecar" images when needed.
You'll need to set `binarySource=docker` for this to work.
Read the [`binarySource` config option docs](https://docs.renovatebot.com/self-hosted-configuration/#binarysource) for more information.
```

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [inline docs/usage/getting-started/running.md:64]:
```suggestion
You must set `binarySource=global`, if you don't then Renovate still installs the latest version of the tools or the requested version from the repository.
```

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [reply docs/usage/getting-started/running.md:67]: I can't find more docs on `containerbase` in our Renovate docs search. Do we have more docs somewhere else?

File: docs/usage/getting-started/running.md
Comment (viceice) [inline docs/usage/getting-started/running.md:214]:
```suggestion
    If you're using Renovate in a project where dependencies are loaded from github.com (such as Go modules hosted on GitHub), we highly recommend that you add a `github.com` PAT (classic).
```

Should we mention classic PAT?

File: docs/usage/getting-started/running.md
Comment (viceice) [inline docs/usage/getting-started/running.md:187]:
```suggestion
We also recommend you configure `config.gitAuthor` with the same identity as your Renovate user, for example: `"gitAuthor": "Renovate Bot <renovate@some.domain.test>"`.
```

Users tend to copy that 😉

File: docs/usage/getting-started/running.md
Comment (viceice) [reply docs/usage/getting-started/running.md:67]: https://docs.renovatebot.com/self-hosted-configuration/#containerbasedir

File: docs/usage/getting-started/running.md
Comment (viceice) [reply docs/usage/getting-started/running.md:62]: Ok, but i think we should still suggest to use the slim image by default

File: docs/usage/getting-started/running.md
Comment (viceice) [reply docs/usage/getting-started/running.md:57]: i think we can use a relative link here, otherwise LGTM

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [inline docs/usage/getting-started/running.md:70]: Do you want me to extract these two lines into a new heading, and put it somewhere suitable?

```markdown
### Persistent cache for downloaded tools

Renovate supports a persistent cache for downloaded tools, so that it only needs to unpack the tools on later runs.
Use the [`containerbaseDir` config option](../self-hosted-configuration.md/#containerbasedir) to control where Renovate stores its containerbase cache.
```

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [reply docs/usage/getting-started/running.md:45]: Thanks for the explanation. ❤️ 

But I still don't know what we want to put in the docs based on your explanation. Can you write a suggestion for me?

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [inline docs/usage/getting-started/running.md:72]:
```suggestion
The default image is for users who don't want to download or install things at runtime.
```

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [inline docs/usage/getting-started/running.md:67]:
```suggestion
Update your Docker images regularly to keep the pre-installed tools up-to-date.
You can use Renovate to update the Renovate Docker images:

MAINTAINERS: INSERT EXAMPLE CONFIG HERE
```

Let's rewrite that sentence, and might be good to add a config example on how to Renovate the Renovate docker images.

Yo dawg, I heard you like Renovate, so I'm Renovating your Renovate docker images with Renovate. 😜

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [inline docs/usage/getting-started/running.md:70]: Commenting to make sure we add the example config. 😉

File: docs/usage/getting-started/running.md
Comment (rarkins) [inline docs/usage/getting-started/running.md:45]:
```suggestion
In general, you can run Renovate natively on Windows as long as you have all tools it will need (e.g. `npm`, `pipenv`, etc.) preinstalled before you run Renovate.
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/getting-started/running.md
Comment (rarkins) [reply docs/usage/getting-started/running.md:68]: Do we definitely need this? If it's in a supported format (e.g. Dockerfile, Docker Compose, GitLab pipelines, etc) then it won't need any manual config

File: docs/usage/getting-started/running.md
Comment (rarkins) [inline docs/usage/getting-started/running.md:68]:
```suggestion
```

File: docs/usage/getting-started/running.md
Comment (rarkins) [reply docs/usage/getting-started/running.md:68]: I think this is maybe to much detail for a getting started. @viceice wdyt?

Comment (viceice) [reply docs/usage/getting-started/running.md:68]: yes, this is something advanced. Maybe we should have a new page about renovate caching. (redis, s3, containerbase, ...)

Comment (HonkingGoose) [reply docs/usage/getting-started/running.md:68]: When I search for `cache` or `caching` in the Renovate docs search I get hits over multiple pages.

Sounds good to have a single page to explain the whole "How Renovate uses caching" thing. I don't know anything about caching. One of the maintainers will have to write a draft that I can review. 😄

Comment (HonkingGoose) [reply docs/usage/getting-started/running.md:68]: @rarkins can you create an issue for us about creating a Renovate caching explanation page, and _what_ it should explain?

I don't know enough to create an actionable issue for you... 🙈

Comment (rarkins) [reply docs/usage/getting-started/running.md:68]: https://github.com/renovatebot/renovate/issues/22284

**Review by rarkins — DISMISSED** [review]

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [inline docs/usage/getting-started/running.md:30]:
```suggestion
By default, Renovate does not support non-npm lock files.
You must install all required third-party tools in the same environment that Renovate's going to run in, _before_ running Renovate.
For example: to use Bundler's `Gemfile.lock` you would first install the Bundler package manager into Renovate's environment.
```

Whew, that was a tricky rewrite. Let's see what you think of this. 😄 

> in the same environment that Renovate's going to run in, _before_ running Renovate.

This part still feels a bit clunky though, open for suggestions!

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [inline docs/usage/getting-started/running.md:209]:
```suggestion
This account can be _any_ account on GitHub, and needs only `read-only` access.
```

**Review by HonkingGoose — COMMENTED** [review]
Two more things and then we're good to go, finally. 😄

File: docs/usage/getting-started/running.md
Comment (rarkins) [reply docs/usage/getting-started/running.md:30]: Let's just say "lock files" and keep it generic - ignoring that npm is usually installed already

Comment (HonkingGoose) [reply docs/usage/getting-started/running.md:30]:
```suggestion
By default, Renovate does not support non-npm lock files.
You must install all required third-party tools in the same environment that Renovate's going to run in, _before_ running Renovate with non-npm lockfiles.
```

Do you mean like this?

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/getting-started/running.md
Comment (rarkins) [inline docs/usage/getting-started/running.md]:
```suggestion
In general, you can run Renovate on Windows with global mode, if you have the required package managers pre-installed.
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [reply docs/usage/getting-started/running.md]: I can't apply your "outdated" suggestion. This is the current text:

> In general, you can run Renovate natively on Windows as long as you have all tools it will need (e.g. `npm`, `pipenv`, etc.) preinstalled before you run Renovate.

Let me know if you want to keep the current text, or rewrite to match your suggestion! 😉

**Review by rarkins — DISMISSED** [review]

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [inline docs/usage/getting-started/running.md:40]: Are you running an image _in_ or _on_ a container?

If it's _in_ here's a suggestion:

```suggestion
You can't run the Docker images in a Windows or macOS container.
```

File: docs/usage/getting-started/running.md
Comment (HonkingGoose) [inline docs/usage/getting-started/running.md:38]: Same _in_ or _on_ question applies here.

**Review by rarkins — APPROVED** [review]

**Comment by HonkingGoose** [comment]
This is good to go now. 🙂

**Pull request merged**
